src/common/income-calc/income-calc.service.ts

```diff
@@ -27,18 +34,28 @@ export class IncomeCalcService {
       }
     }
 
-    return resultIncome * currencyValue;
+    let currencyIncome = resultIncome * currencyValue;
+
+    return (currencyIncome >= 10 ? this.math.round(currencyIncome) : this.math.round(currencyIncome, true)).replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1 ');
   }
 
   public getTimeUnitByCode(units: any[], code: string): any {
-    return units.find(unit => unit.code === code);
+    if (units) {
```

Comment:
@mdmitriy remove it, but add default value or call this function only with arrays (no undefined values)